python/pip_install/pip_repository.bzl

```diff
@@ -766,18 +816,28 @@ def _whl_library_impl(rctx):
     # Manually construct the PYTHONPATH since we cannot use the toolchain here
     environment = _create_repository_execution_environment(rctx, python_interpreter)
 
-    repo_utils.execute_checked(
-        rctx,
-        op = "whl_library.ResolveRequirement({}, {})".format(rctx.attr.name, rctx.attr.requirement),
-        arguments = args,
-        environment = environment,
-        quiet = rctx.attr.quiet,
-        timeout = rctx.attr.timeout,
-    )
+    target_platforms = rctx.attr.experimental_target_platforms
+    platform_agnostic_wheel = _try_finding_platform_agnostic_wheel(rctx, rctx.attr.requirement)
+    if platform_agnostic_wheel and not target_platforms:
```

Comment:
If my understanding is correct, this logic would prefer a pure-python wheel over a native wheel which is not actually what `pip` would do. Pip would prefer a native wheel if one exists for the target platform, and this does occur in a handful of packages that ship native extensions as a performance optimization for certain platforms and a pure-python wheel for the rest ([SQLAlchemy](https://pypi.org/project/SQLAlchemy/2.0.27/#files) does this, for example).